scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/factory/OptionalParserValidator.java

```diff
@@ -25,21 +25,46 @@
  */
 package org.ow2.proactive.scheduler.common.job.factories.spi.model.factory;
 
+import static org.ow2.proactive.scheduler.common.job.factories.spi.model.validator.ModelValidator.OPTIONAL_VARIABLE_SUFFIX;
+
+import org.apache.commons.lang3.StringUtils;
 import org.ow2.proactive.scheduler.common.job.factories.spi.model.converter.Converter;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.converter.NullConverter;
 import org.ow2.proactive.scheduler.common.job.factories.spi.model.exceptions.ModelSyntaxException;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.validator.ModelValidator;
 import org.ow2.proactive.scheduler.common.job.factories.spi.model.validator.OptionalValidator;
-import org.ow2.proactive.scheduler.common.job.factories.spi.model.validator.UserFileValidator;
 import org.ow2.proactive.scheduler.common.job.factories.spi.model.validator.Validator;
 
+import com.google.common.annotations.VisibleForTesting;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 14/08/19
+ */
+public class OptionalParserValidator<T> extends BaseParserValidator<T> {
+    @VisibleForTesting
+    BaseParserValidator<T> parentParserValidator;
 
-public class OptionalUserFileParserValidator extends UserFileParserValidator {
+    public OptionalParserValidator(String model, ModelType type) throws ModelSyntaxException {
+        super(removeSuffix(model), type, ".+");
```

Comment:
I have the feeling this does not allow checking that the model string is valid.
For example, PA:DATETIME([? will produce a not understandable error

Reply:
This line `.+` allows all the string, but during generating the parentParserValidator `parentParserValidator = (BaseParserValidator<T>) ModelValidator.newParserValidator(type, this.model)` (here this.model is the model without suffix `?`), it will get the error `Model expression DATETIME([ doesn't match the expected pattern: ^(?i:DATETIME)\(([^)]+)\)$|^(?i:DATETIME)\(([^)]+)\)\[([^)]+)\]$`.

Reply:
Ok, thanks for the clarification. PR looks very clean.

One word of caution though, as you removed existing PA:OPTIONAL.. I think you should do another PR in ow2-proactive/proactive-examples to change the corresponding workflows which were using these models, and merge them at the same time to avoid issues.

Reply:
Thanks for the reminder, currently none of the example workflows have used the optional variable.
I've just created the PR to remove the front-end UI where have used optional variables and to better handle variable null/undefined case.